devenv.nix

```diff
@@ -7,33 +12,49 @@
     libraries = with pkgs; [
       file # python-magic
     ];
-    poetry = {
+    venv.enable = true; # put venv in PATH
+    uv = {
       enable = true;
-      activate.enable = true;
-      install.enable = true;
-      install = {
-        groups = [ "dev" ];
-        installRootPackage = true;
-      };
+      sync.enable = true;
+      sync.arguments = [
+        # by default it contains `--no-install-workspace`
+        "--frozen"
+        "--group"
+        "dev"
+      ];
     };
   };
 
+  env.UV_LINK_MODE = "copy";
+
   packages =
     with pkgs;
     [
-      nvfetcher
       nodejs # for pyright and renovate
     ]
     ++ unblob.runtimeDeps;
 
   tasks = {
+    "venv:link" = {
+      exec = ''
+        VENV_DIR="${config.devenv.root}/.venv"
+
+        if [[ -d "$VENV_DIR" && ! -L "$VENV_DIR" ]]; then
+          echo "${config.devenv.root}/.venv is a leftover directory. Please remove it."
```

Comment:
If we are sure it is a `leftover` without value, we could just `rm -rf` it. If we are not sure, the message could be softer:
```
echo "${config.devenv.root}/.venv is a directory. Please remove it if you want to use devenv."
```

Reply:
I just don't want to remove anything from a user's computer I don't know about.
I can add the latter part, but the user needn't know about what devenv is and even that we use it under the hood

Reply:
I am fine with it as it is, it is probably in the nitpicking category.

I just found the word `leftover` and the instruction `Please remove it.` a bit too opinionated together.

Reply:
Made it less opinionated